LWA/LWA_bifrost.py

```diff
@@ -844,109 +942,149 @@ def main(self):
+                                        shape=(self.ntime_gulp, nchan, nstand, npol ** 2),
+                                        dtype=numpy.complex64,
+                                        space="cuda",
+                                    )
+                                    autocorrs_av = bifrost.zeros(
+                                        shape=(1, nchan, nstand, npol ** 2),
+                                        dtype=numpy.complex64,
+                                        space="cuda",
+                                    )
+                                    autocorr_g = bifrost.zeros(
+                                        shape=(1, nchan, npol ** 2, self.grid_size, self.grid_size),
+                                        dtype=numpy.complex64,
+                                        space="cuda",
+                                    )
+                                    autocorr_lo = bifrost.ndarray(
+                                        numpy.ones(
+                                            (3, 1, nchan, nstand, npol ** 2),
+                                            dtype=numpy.int32
+                                        ) * self.grid_size / 2,
+                                        space="cuda",
+                                    )
+                                    autocorr_il = bifrost.ndarray(
+                                        numpy.ones(
+                                            (1, nchan, nstand, npol ** 2, self.ant_extent, self.ant_extent),
+                                            dtype=numpy.complex64
+                                        ),
+                                        space="cuda",
+                                    )
 
                                 # Cross multiply to calculate autocorrs
-                                bifrost.map('a(i,j,k,l) += (b(i,j,k,l/2) * b(i,j,k,l%2).conj())',
-                                            {'a':autocorrs, 'b':udata,'t':self.ntime_gulp},
-                                            axis_names=('i','j','k','l'),
-                                            shape=(self.ntime_gulp,nchan,nstand,npol**2))
-
-                            bifrost.map('a(i,j,p,k,l) += b(0,i,j,p/2,k,l)*b(0,i,j,p%2,k,l).conj()',
-                                        {'a':crosspol, 'b':gdata},
-                                        axis_names=('i','j', 'p', 'k', 'l'),
-                                        shape=(self.ntime_gulp, nchan, npol**2, self.grid_size, self.grid_size))
-
+                                bifrost.map(
+                                    "a(i,j,k,l) += (b(i,j,k,l/2) * b(i,j,k,l%2).conj())",
+                                    {"a": autocorrs, "b": udata, "t": self.ntime_gulp},
+                                    axis_names=("i", "j", "k", "l"),
+                                    shape=(self.ntime_gulp, nchan, nstand, npol ** 2),
+                                )
+
+                            bifrost.map(
```

Comment:
You can replace this with the following :

                           ## Cross-Correlation of Antennas
                            if self.benchmark == True:
                                timecorr1 = time.time()

                            try:
                                bf_xcorr.execute(gdata, crosspol)
                            except NameError:
                                bf_xcorr = xCorr()
                                bf_xcorr.init(self.locs,self.grid_size, polmajor=False)
                                bf_xcorr.execute(gdata, crosspol)

                            if self.benchmark == True:
                                timecorr2 = time.time()
                                print(" Cross-Corr time : %f" % (timecorr2 - timecorr1))